app/src/org/commcare/CommCareApplication.java

```diff
@@ -1243,4 +1245,20 @@ public void onStateChanged(@NonNull LifecycleOwner source, @NonNull Lifecycle.Ev
                 break;
         }
     }
+
+    public IDatabase createOrOpenGlobalDatabase() {
+        return new EncryptedDatabaseAdapter(new DatabaseGlobalOpenHelper(this, "null"));
+    }
+
+    public IDatabase createOrOpenUserDatabase(String userKeyRecordId, String key) {
+        return new EncryptedDatabaseAdapter(new DatabaseUserOpenHelper(this, userKeyRecordId, key));
+    }
+
+    public IDatabase openUserDatabase(String path, String password) {
+        return new EncryptedDatabaseAdapter(path, password);
+    }
+
+    public IDatabase createOrOpenAppDatabase(String appId) {
+        return new EncryptedDatabaseAdapter(new DatabaseAppOpenHelper(this, appId, "null"));
+    }

```

Comment:
- I don't really get why do we need to pass params like "null" here. Do they need to be params to Open helpers ?

Optional: rename the methods to `getGlobalDbOpenHelper` and so on as that's what they are actully doing.

Reply:
> I don't really get why do we need to pass params like "null" here. Do they need to be params to Open helpers ?

Not necessarily, I've refactored this.